readme.md

```diff
@@ -586,43 +532,216 @@ const Component = () => {
   ```
 </details>
 
-## Typing your store and `combine` middleware
+## TypeScript Usage
 
-```tsx
-// You can use `type`
-type BearState = {
-  bears: number
-  increase: (by: number) => void
-}
+### Basic usage
+
+When using TypeScript you just have to make a tiny change that instead of writing `create(...)` you'll have to write `create<T>()(...)` where `T` would be type of the state so as to annotate it. Example...
+
+```ts
+import create from "zustand";
 
-// Or `interface`
 interface BearState {
   bears: number
-  increase: (by: number) => void
+  increase: () => 
 }
 
-// And it is going to work for both
-const useStore = create<BearState>(set => ({
+create<BearState>()((set) => ({
   bears: 0,
-  increase: (by) => set(state => ({ bears: state.bears + by })),
+  increase: () => set((state) => ({ bears: state.bears + 1 })),
 }))
 ```
 
-Or, use `combine` and let tsc infer types. This merges two states shallowly.
+<details>
```

Comment:
I think a separate readme for typescript is a little unnecessary, as the "extra info" bits are already collapsed. Having a separate file for typescript makes it look as if the typescript usage is complex, when it's not (it's just the extra infos are a little long). What do you think? Should I still make a separate md for typescript?

> the usage with manual typing

I'd not want to mention this entirely, because it's not recommended. (You see generics in a way are implementation details so they should be as inferred as possible. If the generics are inferred it means we can make changes to the types and keep it non-breaking as long as the inference is the same but if users start passing unnecessary generics even if the inference is the same but a change in type-parameters make it breaking. So it's always recommended to not pass generics as much as possible.)

Also that's for v3 in v4 manual typing would look like this...

```ts
const useStore = create<
  BearState,
  [],
  [
    ['zustand/persist', BearState],
    ['zustand/devtools', never]
  ]
>(devtools(persist((set) => ({
  bears: 0,
  increase: (by) => set(state => ({ bears: state.bears + by })),
})))
```

> We should have Partial<BearState>, right?

No now the default is `BearState` for persist (as it should be)

Reply:
> I think a separate readme for typescript is a little unnecessary, as the "extra info" bits are already collapsed. Having a separate file for typescript makes it look as if the typescript usage is complex, when it's not (it's just the extra infos are a little long). What do you think? Should I still make a separate md for typescript?

Yes, I want to make a separate md.
I plan to move other contents to separate md files too. We want to make readme smaller than what we have now (collapsing is non-ideal workaround).
If you like, "extra info" can be collapsed too in the separate md, as it's very technical.

> > the usage with manual typing
> 
> I'd not want to mention this entirely, because it's not recommended.

I'm not 100% convinced, but we can start without manual typing. I expect some users want it (who don't like `()` workaround, or type inference). Yeah, I'm pretty sure about it. Some like to type store manually. When requested, we could add docs, but I'm afraid you are gone at that time.

> > We should have Partial, right?
> 
> No now the default is `BearState` for persist (as it should be)

Ah, I see. yeah, we changed it. Looks clean.